### DISCUSSION TITLE
How to obtain minecraft time?

### DISCUSSION THREAD

**Discussion by alazeprt** [FabricMC/fabric-api#2981] [Mod Dev Support]
How to obtain minecraft time?

**Comment by warjort** [comment]
Your question is unclear.
Maybe? `World.getTime()` or `World.getTimeOfDay()`

**Comment by alazeprt** [comment]
I want to modify the block state by obtaining time. During the day, it is true, and at night, it is false.

**Reply by warjort** [reply]
World.isDay() or World.isNight()

**Reply by alazeprt** [reply]
What method do I use to modify the status?

**Reply by warjort** [reply]
Sorry, but you can't expect support if your questions are so vague. 
We have no psychic powers. :-)

**Reply by alazeprt** [reply]
I want the block to automatically change its status to true during the day, and to automatically change its status to false at night. How do I achieve this?

**Reply by warjort** [reply]
Look at the vanila DaylightDetectorBlock[Entity] classes.

**Reply by alazeprt** [reply]
In my block entity class, I used the tick method to modify the block state:
<code>
    public static void tick(World world, BlockPos pos, BlockState state, solar_panel_entity be) {
        if(world.isDay()){
            world.setBlockState(pos, state.with(DAYTIME, 1), 3);
        } else{
            world.setBlockState(pos, state.with(DAYTIME, 0), 3);
        }
    }</code>
However, at night, the value of the DAYLIGHT status will continuously change. How can I modify it?
(I tried to output the value of world.isDay() in the log):
In the morning:
<code>[09:04:44] [Render thread/INFO]: true
[09:04:44] [Server thread/INFO]: true
[09:04:44] [Render thread/INFO]: true
[09:04:44] [Server thread/INFO]: true
[09:04:44] [Render thread/INFO]: true
[09:04:44] [Server thread/INFO]: true
</code>
At night:
<code>[09:04:50] [Render thread/INFO]: true
[09:04:50] [Server thread/INFO]: false
[09:04:50] [Render thread/INFO]: true
[09:04:50] [Server thread/INFO]: false
[09:04:50] [Render thread/INFO]: true
[09:04:50] [Server thread/INFO]: false
</code>
